### PR TITLE
Bootstrap version updated to remove jQuery dependency

### PR DESCRIPTION
**Pull request by AritraLeo**
These are my initial changes so far and I'd am open to update any of the other files in case it's needed. 

Before opening this PR I have - 

> performed e2e testing for chrome and edge 

> also build the project locally

Please have a look !  
issue - https://github.com/kiwix/kiwix-js/issues/1233

### CODE REVIEW

**Comment by Jaifroid** [comment]
Thank you! I'm just running automated tests.

**Comment by AritraLeo** [comment]
> Thank you! I'm just running automated tests.

Sure also please let me know if any changes are required, this is my first PR to your project I am just being cautious!

**Comment by Jaifroid** [comment]
Looking good so far! I see this error in console.log, so you need to remove JQuery from the Service Worker's precache array.

image

But, more seriously, when the screen is narrow and the menus have collapsed, the toggle no longer seems to be working. You probably have to adjust the commands in `app.js` (possibly `uiUtil.js`) to whatever new syntax is needed for Bootstrap 5.

**Comment by Jaifroid** [comment]
NB to see the precache error, you probably have to reset the app, so it erases the SW and then reloads it.

**Comment by Jaifroid** [comment]
We probably need to tweak the styles of the buttons, because hovering over them now makes them greyed out rather than darker as previously, which looks odd.

**Comment by AritraLeo** [comment]
> We probably need to tweak the styles of the buttons, because hovering over them now makes them greyed out rather than darker as previously, which looks odd.

Ohk I assume this is due to newer version of bootstrap have to tweak the styling and all. 
But can you please provide me with little more info about - 
"NB to see the precache error, you probably have to reset the app, so it erases the SW and then reloads it."

**Comment by Jaifroid** [comment]
It's explained in CONTRIBUTING -- the app caches its own code (in the Service Worker). To see if the error is solved, you'll need to use the Reset button (under Expert Settings) to erase the old Service Worker and the Cache, and then watch in console.log as it repopulates the Cache according to the code in ´service-worker.js`. There are other ways to do this, like forcing a reload of the Service Worker on refresh (an option under Application in Dev Tools). Note that after resetting the app, you should turn on the option to Bypass AppCache again (Expert Settings).

**Comment by AritraLeo** [comment]
@Jaifroid 
Thank you so much for such a nice a response really appreciate it! 
I'll surely update the PR with the necessary changes and let you know.

**Comment by AritraLeo** [comment]
@Jaifroid I found the issue for the toggle actually there was nothing to change in app/uUtil.js but bootstrap uses it's own attrs now with **bs**. Like data-bs-toggle. And removed the jQuery from precache running tests now (also setup my dev env to update cache on reload - I skipped that part on CONTRIBUTING file my bad) 

If there's anything else please let me know !

**Comment by Jaifroid** [comment]
> Is this ideal ?

Did you test it in dark mode (and you should probably turn off the bypass appCache option when testing the buttons at least in dark mode, as it tends to paint things red)? My initial reaction is that a somewhat more subtle colour change would be preferable to the button turning black! Maybe a somewhat darker grey instead?

**Comment by AritraLeo** [comment]
> > I found the issue for the toggle actually there was nothing to change in app/uUtil.js but bootstrap uses it's own attrs now with **bs**. Like data-bs-toggle. And removed the jQuery from precache running tests now (also setup my dev env to update cache on reload - I skipped that part on CONTRIBUTING file my bad). Also checked the button hover thing.
> > If there's anything else please let me know !
> 
> Great! Could you push your code so I can test? You also might want to update the branch (see "Update branch" button below), and don't forget to pull locally so that the update gets added to your PR before pushing more code.

Sure I'll push my code will give the button issue a check in dark mode and push my changes, and ya I mostly update my branch and pull locally but hey thanks for the tip appreciate it!

**Comment by AritraLeo** [comment]
> > I found the issue for the toggle actually there was nothing to change in app/uUtil.js but bootstrap uses it's own attrs now with **bs**. Like data-bs-toggle. And removed the jQuery from precache running tests now (also setup my dev env to update cache on reload - I skipped that part on CONTRIBUTING file my bad). Also checked the button hover thing.
> > If there's anything else please let me know !
> 
> Great! Could you push your code so I can test? You also might want to update the branch (see "Update branch" button below), and don't forget to pull locally so that the update gets added to your PR before pushing more code.

Actually I have used Bootstrap 5's btn-dark because others were not apt. I didn't tweak the styling but ya sure I can.
Just let me know the hex code you want for hover I'll try to incorporate it.

**Comment by AritraLeo** [comment]
@Jaifroid 
In dark mode - 
image
Without hover

image
On hover.

**Comment by AritraLeo** [comment]
@Jaifroid please share your thoughts!

**Comment by Jaifroid** [comment]
Thanks for the changes, and apologies for the delay (I was away travelling). I like the colour of the button you changed, but there's something wrong with the sizing of it compared to the other buttons, and it appears greyed out when not selected, which I don't think is a good idea:

image

I also noticed some differences between the Bootstrap 5 version and the original version, which I've circled in the side-by-side comparison below. Could you please try to remediate these? The spacing of the word "Kiwix" looks particularly bad, as it's lost its margin.

image

**Comment by AritraLeo** [comment]
Hey @Jaifroid I've made the necessary changes so far but I am afraid that complete migration to Bootstrap 5 may require even more changes. In case you find something else please let me know I'll try to resolve any other issue asap. 
Thank You!

**Comment by Jaifroid** [comment]
Hi @AritraLeo, a lot of things are fixed here, but I noticed a new problem: the search field has now become very short! See screenshot:

image

You should look into fixing this with Bootstrap methods ideally rather than just forcing a new style of the search bar in app.css, though of course you should check if an existing style rule exists in app.css that may be conflicting with Bootstrap 5.

**Comment by AritraLeo** [comment]
@Jaifroid  I was concerned about these minor bugs which will pop up anyway. So I think I'll work on it for now but I'll just add it to my doc as well .

**Comment by Jaifroid** [comment]
OK, no problem. I suggest you update the branch (you should see "Update branch" button below).

**Comment by AritraLeo** [comment]
> OK, no problem. I suggest you update the branch (you should see "Update branch" button below).

Sure I'll (saw an 86 issues it was 87 yesterday knew something got resolved) Thanks for the tip

**Comment by Jaifroid** [comment]
There's a thing I forgot to mention that needs doing: remove jquery from the `patch_gitignore.sh` script under `/scripts`. Basically, just remove any of the lines that have `jquery` anywhere in the line. To understand what this does, look at `ADDING_DEPENDENCIES_NODE_MODULES.md`.

**Comment by AritraLeo** [comment]
> There's a thing I forgot to mention that needs doing: remove jquery from the `patch_gitignore.sh` script under `/scripts`. Basically, just remove any of the lines that have `jquery` anywhere in the line. To understand what this does, look at `ADDING_DEPENDENCIES_NODE_MODULES.md`.

Sure! I'll get it done.

**Comment by AritraLeo** [comment]
> Hi @AritraLeo, a lot of things are fixed here, but I noticed a new problem: the search field has now become very short! See screenshot:
> 
> image
> 
> You should look into fixing this with Bootstrap methods ideally rather than just forcing a new style of the search bar in app.css, though of course you should check if an existing style rule exists in app.css that may be conflicting with Bootstrap 5.

Hey @Jaifroid 

I am unable to find this issue in my setup 
Is there any specific reason for that ? 

image

**Comment by Jaifroid** [comment]
@AritraLeo I think you need to fully refresh your Service Worker to ensure your browser is showing the latest code. You can test by opening an "In Private" or "Incognito" window and loading the app in there. I'm seeing the issue in Edge and in Firefox.

image

**Comment by AritraLeo** [comment]
> @AritraLeo I think you need to fully refresh your Service Worker to ensure your browser is showing the latest code. You can test by opening an "In Private" or "Incognito" window and loading the app in there. I'm seeing the issue in Edge and in Firefox.
> 
> image

Sure I'll try it in Firefox with Hard Reload let's see.

**Comment by AritraLeo** [comment]
@Jaifroid 
I am still unable to find this issue (using Edge) 
Can you please provide me with a way to refresh the cache maybe in some other way? But I don't think there's any issue with cache because I am running the server again after executing a Hard reload. 

image

**Comment by Jaifroid** [comment]
For starters, you don't have the "Bypass AppCache" setting turned on. This is necessary. Under the Network tab, did you make sure "Disable cache" is turned on as advised in CONTRIBUTING? With DevTools open, use Ctrl-Shift-R to hard-refresh. This always works with Chrome / Edge if Bypass AppCache is on and caching disabled. Firefox is trickier, I'm afraid, and you might have to delete the Service Worker manually (in Dev Tools -> Application tab) and also the two Cache API Caches.

However, as I mentioned earlier, a sure-fire way to see the issue without doing any of those things is simply to put your browser into InPrivate / Incognito / "new private window" mode. Then load the localhost address. Did you try that? In Firefox, you can tell if you've enabled this (see screenshot):

image

**Comment by AritraLeo** [comment]
> For starters, you don't have the "Bypass AppCache" setting turned on. This is necessary. Under the Network tab, did you make sure "Disable cache" is turned on as advised in CONTRIBUTING? With DevTools open, use Ctrl-Shift-R to hard-refresh. This always works with Chrome / Edge if Bypass AppCache is on and caching disabled. Firefox is trickier, I'm afraid, and you might have to delete the Service Worker manually (in Dev Tools -> Application tab) and also the two Cache API Caches.
> 
> However, as I mentioned earlier, a sure-fire way to see the issue without doing any of those things is simply to put your browser into InPrivate / Incognito / "new private window" mode. Then load the localhost address. Did you try that? In Firefox, you can tell if you've enabled this (see screenshot):
> 
> image

Yeah I tried with Hard Reload and all the things you've mentioned I'll just try all of this once more and delete the worker file this  time I think it'll work

**Comment by Jaifroid** [comment]
If you tried all the things mentioned, and you tried the browser's Private mode, then I can only conclude we are not testing the same code. Perhaps your local code has not been pushed fully to this branch. Ensure your local branch is not dirty (no uncommitted changes), and ensure you have pulled/pushed the branch before testing again. I'm consistently seeing the error on both Firefox and Chrome. I also suggest you update the branch again and push/pull.

**Comment by AritraLeo** [comment]
> If you tried all the things mentioned (though Bypass AppCache was not turned on in your screenshot), and you tried the browser's Private mode, then I can only conclude we are not testing the same code. Perhaps your local code has not been pushed fully to this branch. Ensure your local branch is not dirty (no uncommitted changes), and ensure you have pulled/pushed the branch before testing again. I'm consistently seeing the error on both Firefox and Chrome. I also suggest you update the branch again and push/pull, and test in Chrome/Edge as well as Firefox.

Yeah I know I am going wrong somewhere. Just trying to find it.
Anyways I'll get back to you with the issue resolved this time !

**Comment by Jaifroid** [comment]
@AritraLeo As mentioned on Slack, I've discovered that the error is an artefact of my not having re-updated the software with `npm install` after switching branches to review someone else's work. Hence, my system was loading an older version of Bootstrap when I switched back to your branch.

Having corrected that, the short search bar issue has gone. I apologize for that oversight.

**Comment by AritraLeo** [comment]
> @AritraLeo As mentioned on Slack, I've discovered that the error is an artefact of my not having re-updated the software with `npm install` after switching branches to review someone else's work. Hence, my system was loading an older version of Bootstrap when I switched back to your branch.
> 
> Having corrected that, the short search bar issue has gone. I apologize for that oversight.

Yeah! I see no issues with that.

File: www/css/app.css [hunk 1]
```diff
@@ -294,6 +294,17 @@ button {
...
+
+.btn-outline-dark:hover {
+    color: #fff;
+    background-color: #6c757d;
+    border-color: #6c757d;
+}
+
+
+
+
```
Comment (Jaifroid) [inline www/css/app.css:307]: Close up the extraneous space here (just leave one space).

File: www/index.html [hunk 2]
```diff
@@ -465,15 +465,17 @@ <h2 data-i18n="configure-title">Configuration</h2>
                             <button type="button" class="close" data-hide="alert">&times;</button>
                             <span id="persistentMessage"></span>
                         </div>
+                        <!-- Dropdown prev (updated but arrow isn't visible) -->
                         <div class="dropdown" style="float:right">
-                            <select class="form-control" tabindex="4" id="languageSelector" name="languageSelector" aria-labelledby="languageSelector">
-                                <option data-i18n="configure-language-selector-default" selected class="dropdown-item" href="#" value="default">Language</a>
+                            <select class="form-select dropdown-toggle" tabindex="4" id="languageSelector" data-bs-toggle="dropdown" name="languageSelector" aria-labelledby="languageSelector">
+                                <option data-i18n="configure-language-selector-default" selected class="dropdown-item" href="#" value="default">Language </a>
```
Comment (Jaifroid) [inline www/index.html:471]: Remove extraneous space after Language.

File: www/index.html [hunk 3]
```diff
@@ -465,15 +465,17 @@ <h2 data-i18n="configure-title">Configuration</h2>
...
                                 <option class="dropdown-item" href="#" value="en">English</a>
                                 <option class="dropdown-item" href="#" value="fr">Français</a>
                                 <option class="dropdown-item" href="#" value="es">Español</a>
                                 <option data-i18n="configure-language-selector-other" class="dropdown-item" href="#" value="other">More coming soon...!</a>
                             </select>
                         </div>
+
```
Comment (Jaifroid) [inline www/index.html:478]: Remove extraneous empty line.

File: www/index.html [hunk 4]
```diff
@@ -800,7 +802,7 @@ <h3 data-i18n="configure-expert-settings-title">Expert settings</h3>
     <!-- Legacy jQuery and Bootstrap are not ES-module-aware, and interact with each other using side-effects, so we are forced to load them globally here.
         There is code in rollup.config.js which alters these references, and the link reference above, to point to the local destination of the copied modules.
     -->
-    <script type="text/javascript" src="../node_modules/jquery/dist/jquery.slim.min.js"></script>
+    <!-- <script type="text/javascript" src="../node_modules/jquery/dist/jquery.slim.min.js"></script> -->
```
Comment (Jaifroid) [inline www/index.html:805]: You can remove this comment now.

**Review by Jaifroid — CHANGES_REQUESTED** [review]
@AritraLeo Sorry for the hiatus. Getting back to this PR, here are some very minor code-format things, and I'll list the other non-code issues I'm seeing in the next comment.

**Comment by AritraLeo** [comment]
I'll get them done @Jaifroid.

File: rollup.config.js [hunk 5]
```diff
@@ -131,7 +133,7 @@ if (process.env.BUILD === 'production') {
                     // Comment out the old app.js link
                         .replace(/(<script type="module.*app.js.*)/, '<!-- $1 -->')
                     // Redirect jQuery and bootstrap
-                        .replace(/(<script\s.*src=").*jquery.slim.min.js/, '$1js/jquery.slim.min.js')
+                        // .replace(/(<script\s.*src=").*jquery.slim.min.js/, '$1js/jquery.slim.min.js')
```
Comment (Jaifroid) [inline rollup.config.js:136]: Remove commented-out line.

File: rollup.config.js [hunk 6]
```diff
@@ -184,7 +186,7 @@ if (process.env.BUILD === 'production') {
...
```
Comment (Jaifroid) [inline rollup.config.js:189]: Remove.

**Comment by Jaifroid** [comment]
Here are some other issues I noticed:

1. There are some oddities on the home screen (see screenshot). Most important here is that the icons on the bottom bar are not correctly spaced as they are in the currently released version;
2. The spacing generally is too compacted in Configuration. You can compare side-by-side the currently released browser extension and this PR, and you'll see what I mean. We have to have adequate spacing to cater for touchscreen support;
3. You still need to remove jQuery from: `Check-OfflineFilesList.ps1` and `patch_gitignore.sh`;
4. There's a dangling comma in `service-worker.js` line 153;
5. Please update the branch to the latest.

image

File: www/index.html (hunk 3 above)
Comment (AritraLeo) [reply www/index.html:478]: Linter on my system causing this on hitting save the format automatically takes it next line.

**Comment by AritraLeo** [comment]
@Jaifroid Hey so sorry for having no updates on PR (you know why). I have got the none code issues resolved have a look I'll start with the other issue mentioned and get the tasks done asap. Thanks!

**Comment by Jaifroid** [comment]
Apart from resolving the conflicts, take a look at the reasons for the Codefactor failure: https://github.com/kiwix/kiwix-js/pull/1235/checks?check_run_id=25892549156 . They're usually just formatting fixes.

**Comment by Jaifroid** [comment]
It's looking a lot better now, and is nearly there! There is just the issue of overly compact spacing in the version in your PR. This is minor, but we have to bear in mind that the app is used also on touchscreen devices. I've drawn some lines for you to compare the spacing at the connected points.

This should be easy to fix, either by wrapping some elements in paragraphs, or by adding appropriate spacing for the affected elements in `app.css`. Then we'll be pretty much there, I think!

**Comment by AritraLeo** [comment]
@Jaifroid I'll just check with Bootstrap v5 docs again if there's some class name that's deprecated or not. Else make changes to the app.css file. I'll update the branch as per required and get back to you.

**Comment by Jaifroid** [comment]
Feel free to re-open if you want to complete this.

**Comment by AritraLeo** [comment]
> Feel free to re-open if you want to complete this.

I am extremely sorry for taking so long. I want to get back to this PR tomorrow morning and get the changes done (the merge to be precise). Apologies again. (Reason in your dm)

**Comment by Jaifroid** [comment]
@AritraLeo Thank you very much for continuing work on the PR. For some reason, tests are looking pretty broken, and not just with the usual timeout issue. See https://github.com/kiwix/kiwix-js/actions/runs/10201928029/job/28225674553?pr=1235 and https://github.com/kiwix/kiwix-js/actions/runs/10201928029/job/28225674843?pr=1235. You can debug locally with `npm test`, or open `tests/index.html` in the browser you want to test. See https://github.com/kiwix/kiwix-js/blob/main/TESTS.md for more info.

**Comment by Jaifroid** [comment]
Sorry, that command was for Unit tests. But you probably need to debug e2e tests as well. See https://github.com/kiwix/kiwix-js/blob/main/TESTS.md#end-to-end-tests. To test e2r on Chrome, it's `npm run test-e2e-chrome`.

**Comment by AritraLeo** [comment]
> Sorry, that command was for Unit tests. But you probably need to debug e2e tests as well. See https://github.com/kiwix/kiwix-js/blob/main/TESTS.md#end-to-end-tests. To test e2r on Chrome, it's `npm run test-e2e-chrome`.

Ohk so I'll have to check why the e2e tests are failing right? Only this one?

**Comment by Jaifroid** [comment]
> Ohk so I'll have to check why the e2e tests are failing right? Only this one?

Well you could run local tests in Firefox if you prefer. `npm run test-e2e-firefox`. The Firefox test didn't run because the Chrome test failed. I'll bet it encounters the same issue, whatever it is.

**Comment by Jaifroid** [comment]
This PR was passing tests prior to https://github.com/kiwix/kiwix-js/pull/1235/commits/8f91d7a41aec3190008e1b073f73ab025304fb88, so looking at the code you changed there might give you a clue as to why tests are failing consistently since.

**Comment by AritraLeo** [comment]
> This PR was passing tests prior to [8f91d7a](https://github.com/kiwix/kiwix-js/commit/8f91d7a41aec3190008e1b073f73ab025304fb88), so looking at the code you changed there might give you a clue as to why tests are failing consistently since.

Yeah sure I'll look into it I was not in good health recently and also got caught up in my job. Spare me a but more time. Apologies for taking so long again!

**Comment by Jaifroid** [comment]
Oh, sorry to hear that, no problem of course!

**Comment by AritraLeo** [comment]
> Oh, sorry to hear that, no problem of course!

Please check dm in slack
